ReactQt/runtime/src/reactuimanager.cpp

```diff
@@ -128,8 +128,8 @@ void ReactUIManager::setChildren
 {
   //TODO: This is a simple implementation which fixes a broken example. It's not properly tested and may need revisiting
   QList<int> indices;
-  foreach(int i, childrenTags) {
-    indices.append(0); //0 is a default z-order in qml
+  for(int i = 0; i < childrenTags.size(); ++i) {
+    indices.append(i); //Incrementing per each child. Similar to iOS. See the comment in manageChildren
```

Comment:
Hi @narek-h , mentioned comment seems just means that Zposition of UIView on iOS is similar functionality as Z order in QML. But I was not able to find in iOS implementation explicit setting of Zposition in react-native implementation. This also should mean that we shouldn't do it as well.
It might be great to look into possibility to support react-native zIndex property in scope of this bug, or create separate issue for this. Adding support of zIndex and covering with some tests should resolve current issue.